feat: restrict new configuration parameters

higor-melo opened the pull request:
# What is this PR for
<!-- A clear description of what this PR do and the changes made.
If you want you can also add the shortcut link here
- [ ] [Shortcut](http://) -->
The firmware on the current installed camera is outdated therefore it doesn't allow the configuration of these new parameters without bugging randomly.

# Results, How I tested
<!-- Explain how you tested you PR, if there is a visual output,
     a GIF or image is welcome -->
Not tested yet but it's pretty straight foward, will test ASAP

# Checklist
- [x] Assign yourself  in the PR

higor-melo committed:
feat: restrict new configuration parameters

The firmware on the current installed camera
is outdated therefore it doesn't allow the
configuration of these new parameters without
bugging randomly.

DeboraTahara commented on tasks/Task.cpp:
```diff
@@ -1008,21 +1015,23 @@ void Task::balanceConfiguration(Arena::IDevice& device)
-        _analog_controller_config.get().balance_white_enable);
+        Arena::SetNodeValue<bool>(device.GetNodeMap(),
+            "BalanceWhiteEnable",
+            _analog_controller_config.get().balance_white_enable);
```
trailing spaces

DeboraTahara commented on tasks/Task.cpp:
```diff
@@ -1082,27 +1091,35 @@ void Task::balanceConfiguration(Arena::IDevice& device)
-        Arena::SetNodeValue<gcstring>(device.GetNodeMap(),
-            "BalanceRatioSelector",
-            "Green");
+        if (!base::isUnknown(_analog_controller_config.get().green_balance_ratio)) {
```
I really recommend you to save the property value in a variable instead of doing many gets

higor-melo committed:
chore: add analog_config variable

to avoid using too many .get()

higor-melo committed:
fix: base::unknown doesn't exist for booleans

higor-melo committed:
feat: add new test

This test was used to verify the behavior
after multiple reconfigurations which
caused a crash from a buffer timeout
